gpflow/logdensities.py

```diff
@@ -85,17 +83,16 @@ def multivariate_normal(x, mu, L):
     p[n] = log pdf of:
     x[n] ~ N(mu, LL^T) or x ~ N(mu[n], LL^T) or x[n] ~ N(mu[n], LL^T)
     """
-    mu_ndims = mu.shape.ndims
-    x_ndims = x.shape.ndims
-    if x_ndims is not None and x_ndims != 2:
-        raise ValueError('Shape of x must be 2D.')
-    if mu_ndims is not None and mu_ndims != 2:
-        raise ValueError('Shape of mu must be 2D.')
-
-    d = x - mu
-    alpha = tf.matrix_triangular_solve(L, d, lower=True)
-    num_dims = tf.cast(tf.shape(d)[0], L.dtype)
-    p = - 0.5 * tf.reduce_sum(tf.square(alpha), 0)
-    p -= 0.5 * num_dims * np.log(2 * np.pi)
-    p -= tf.reduce_sum(tf.log(tf.matrix_diag_part(L)))
-    return p
+
+    assert_x_shape = tf.Assert(tensor_ndim_equal(x, 2), [
+        'multivariate_normal requires the shape of x to be (N, D)'])
+    assert_mu_shape = tf.Assert(tensor_ndim_equal(mu, 2), [
+        'multivariate_normal requires the shape of mu to be (N, D)'])
+    with tf.control_dependencies([assert_x_shape, assert_mu_shape]):
```

Comment:
For some reason the asserts don't actually get triggered...

Reply:
because I tested with constants, not with placeholders fed in through the feed_dict...